src/main/scala/io/viash/functionality/dependencies/Repository.scala

```diff
@@ -145,14 +145,17 @@ object Repository {
     |""".stripMargin,
   "yaml"
   )
-case class GithubRepository(
+case class GitRepository(
   name: String,
 
-  @description("Defines the repository as a GitHub repository.")
+  @description("Defines the repository as a Git repository.")
   `type`: String = "github",
 
-  @description("The GitHub `organization/repository_name` of the repository.")
-  @example("uri: viash-io/viash", "yaml")
+  @description(
+    """For github: The GitHub `organization/repository_name` of the repository.
+      |For git: The Git `url:organization/reporistory_name` of the repository.
```

Comment:
I'd consider making a GitHub repository a separate case class from GitRepository. But these classes would share a lot of functionality, so maybe:

```scala
abstract class AbstractGitRepository {
  ...
}
case class GitRepository(...) extends AbstractGitRepository
case class GitHubRepository(...) extends AbstractGitRepository
```

WDYT?